docs/controllers.md

```diff
@@ -0,0 +1,152 @@
+flow is in the opposite direction. Two elements are important to be aware of:
+
+- Interchain Security Modules (ISMs): these modules are used to verify a received message.
+- Post dispatch hooks: these modules are used to execute post dispatch logic when a message is sent.
+
+For additional information, please refer to the [Hyperlane docs](https://docs.hyperlane.xyz/).
+
+The Hyperlane controller is the specific implementation of the controller interface designed to
+handle Hyperlane transfers. A Hyperlane transfer is defined by the
+[`HypAttributes`](https://github.com/noble-assets/orbiter/blob/main/proto/noble/orbiter/controller/forwarding/v1/hyperlane.proto#L12-L48)
+type, which is one of the concrete implementations of the forwarding attributes interface. Below is
+reported the proto implementation of the attributes:
+
+```protobuf
+message HypAttributes {
+  option (cosmos_proto.implements_interface) = "orbiter.core.v1.ForwardingAttributes";
+
+  bytes token_id = 1;
+  uint32 destination_domain = 2;
+  bytes recipient = 3;
+  bytes custom_hook_id = 4;
+  string gas_limit = 5 [
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true,
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int"
+  ];
+  cosmos.base.v1beta1.Coin max_fee = 6 [
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true,
+    (amino.encoding) = "legacy_coins",
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+  string custom_hook_metadata = 7;
+}
+```
+
+When the controller receives the forwarding packet, the following steps are executed:
+
+- Hyperlane attributes are extracted from the generic core forwarding type.
+- Hyperlane and transfer attributes are validated.
+- The forwarding is executed.
+
+```mermaid
+---
+title: Hyperlane Controller Workflow Diagram
+---
+
+flowchart LR
+   subgraph Orbiter Module
+     direction LR
+      F[Forwarder]
+      HC[[Hyperlane Controller]]
+      EA(Extract attributes)
+      VF(Validate forwarding)
+   end
+
+  F -- forwarding payload --> HC
+  HC -.-> EA
+  EA -.->|HypAttributes| HC
+  HC -.-> VF
+  VF -.->|bool| HC
+  HC -- remote transfer--> W[Warp Module]
+```
+
+The validation step consists of basic validation on the types, like checks for nil pointers, and the
+check on the specified token ID. This validation uses the Warp application query server to retrieve
+the `WrappedHypToken`, and then verifies that the associated origin denom is the same as the
+destination denom specified in the transfer attributes. Remember that a denom can be associated with
+multiple token IDs with Hyperlane, so it is of paramount importance to specify the correct
+identifier.
+
+For the forwarding to work, the Hyperlane token ID must exist and it has to be enrolled with a
+router associated with the destination chain. This router is used to retrieve the contract on the
```

Comment:
maybe here it could be good to e.g. link this stuff to the tokens query for the warp module?
https://api.testnet.noble.xyz/#tag/warp/GET/hyperlane/v1/tokens